lte/gateway/c/oai/lib/pipelined_client/PipelinedServiceClient.h

```diff
@@ -0,0 +1,212 @@
+      std::function<void(grpc::Status, UESessionContextResponse)> callback);
+
+  static int UpdateUEIPv4v6SessionSetWithFlowdl(
+      struct in_addr& enb_ipv4_addr, const struct in_addr& ue_ipv4_addr,
+      struct in6_addr& ue_ipv6_addr, uint32_t in_teid, uint32_t out_teid,
+      const struct ip_flow_dl& flow_dl, uint32_t state,
+      std::function<void(Status, UESessionContextResponse)> callback);
+
+  // APIs for discarding data on tunnels
+  static int UpdateUEIPv4SessionSet(
+      const struct in_addr& ue_ipv4_addr, uint32_t in_teid, uint32_t state,
+      std::function<void(grpc::Status, UESessionContextResponse)> callback);
+
+  static int UpdateUEIPv4SessionSetWithFlowdl(
+      const struct in_addr& ue_ipv4_addr, uint32_t in_teid,
+      const struct ip_flow_dl& flow_dl, uint32_t state,
+      std::function<void(Status, UESessionContextResponse)> callback);
+
+  static int UpdateUEIPv4v6SessionSet(
+      const struct in_addr& ue_ipv4_addr, struct in6_addr& ue_ipv6_addr,
+      uint32_t in_teid, uint32_t state,
+      std::function<void(grpc::Status, UESessionContextResponse)> callback);
+
+  static int UpdateUEIPv4v6SessionSetWithFlowdl(
+      const struct in_addr& ue_ipv4_addr, struct in6_addr& ue_ipv6_addr,
+      uint32_t in_teid, const struct ip_flow_dl& flow_dl, uint32_t state,
+      std::function<void(Status, UESessionContextResponse)> callback);
+
+  // APIs for forwarding data on the tunnel
+  static int UpdateUEIPv4SessionSet(
+      const struct in_addr& ue_ipv4_addr, uint32_t in_teid,
+      uint32_t flow_precedence, uint32_t state,
+      std::function<void(grpc::Status, UESessionContextResponse)> callback);
+
+  static int UpdateUEIPv4SessionSetWithFlowdl(
+      const struct in_addr& ue_ipv4_addr, uint32_t in_teid,
+      const struct ip_flow_dl& flow_dl, uint32_t flow_precedence,
+      uint32_t state,
+      std::function<void(Status, UESessionContextResponse)> callback);
+
+  static int UpdateUEIPv4v6SessionSet(
+      const struct in_addr& ue_ipv4_addr, struct in6_addr& ue_ipv6_addr,
+      uint32_t in_teid, uint32_t flow_precedence, uint32_t state,
+      std::function<void(grpc::Status, UESessionContextResponse)> callback);
+
+  static int UpdateUEIPv4v6SessionSetWithFlowdl(
+      const struct in_addr& ue_ipv4_addr, struct in6_addr& ue_ipv6_addr,
+      uint32_t in_teid, const struct ip_flow_dl& flow_dl,
+      uint32_t flow_precedence, uint32_t state,
+      std::function<void(Status, UESessionContextResponse)> callback);
+
+  // APIs for paging IDLE -> PAGING, ACTIVATE -> Delete PAGING
+  static int UpdateUEIPv4SessionSet(
+      const struct in_addr& ue_ipv4_addr, uint32_t state,
+      std::function<void(grpc::Status, UESessionContextResponse)> callback);
+
+ public:
+  static PipelinedServiceClient& get_instance();
+  PipelinedServiceClient(PipelinedServiceClient const&) = delete;
+  void operator=(PipelinedServiceClient const&) = delete;
+
+  // Set the UE IPv4 address
+  static void ue_set_ipv4_addr(
```

Comment:
These functions that set values into proto structs should all be moved out into the `proto_converters.cpp` file. (I think you might have moved most of them, so maybe you just have to remove them from this class.)

Reply:
Comments addressed.